state/protocol/badger/mutator.go

```diff
@@ -365,6 +365,12 @@ func (m *FollowerState) headerExtend(ctx context.Context, candidate *flow.Block,
 				return fmt.Errorf("could not store incorporated qc: %w", err)
 			}
 		} else {
+			// parent is a block that has been received and certified by a QC.
+			err := transaction.WithTx(operation.SkipDuplicates(operation.IndexCertifiedBlockByView(parent.View, qc.BlockID)))(tx)
```

Comment:
@AlexHentschel I was using the db operation directly, for 2 reasons:

I wanted to add a IndexCertifiedBlockByView method to headers, and simply call m.headers.IndexCertifiedBlockByView(parent) here. But the storage.Headers interface is mostly for read methods, we didn't expose methods like IndexFinalizedBlockByView because we didn't want this method being misused.
For the same reason I also didn't add IndexCertifiedBlockByView method to Blocks, and also, the parent block body doesn't exist in this context, passing a block to blocks.IndexCertifiedBlockByView would require us to make additional query to the parent block body.